documentation/projects/proposals/rekognition_data/project_proposal.md

```diff
@@ -0,0 +1,142 @@
+
+This project will be accomplished in two major pieces:
+
+1. Determining how machine-generated tags will be displayed/conveyed in the API
+   and the frontend
+2. Augmenting the catalog database with the tags we deem suitable
+
+Focusing on the frontend first may seem like putting the cart before the horse,
+but it seems prudent to imagine how the _new_ data we add will show up in both
+the frontend and the API. While both of the above will be expanded on in
+respective implementation plans, below is a short description of each piece.
+
+### Machine-generated tags in the API/Frontend
+
+The API's [`tags` field][api_tags_field] already has a spot for `accuracy`,
+along with the tag `name` itself. This is where we should include the label
+accuracy that Rekognition provides alongside the label. We should also consider
+including a new `source` section within the array of tag objects, in order to
+communicate where this accuracy value came from. In the future, we may have
+multiple instances of the same label with different `source` and `accuracy`
+values (for instance, if we chose to apply multiple machine labeling processes
+to our media records).
+
+_NB: I'm not sure if this change to the API response shape for `tags` would
+constitute an API version change. I do think having a mechanism to share tag
+source will be important going forward._
+
+[api_tags_field]:
+  https://api.openverse.engineering/v1/#tag/images/operation/images_search
+
+We should also distinguish the machine-generated tags from the creator-added
+ones in the frontend. Particularly with the introduction of the
+[additional search views](../additional_search_views/index.md), we will need to
+consider how these machine-generated tags are displayed and whether they can be
+interacted with in the same way. Similar to the API, it may also be useful to
+share the label accuracy with users (either visually or with extra content on
+mouse hover).
+
+None of the above is specific to Rekognition, but it will be necessary to
+determine for Rekognition or any other labels we wish to add in the future.
+
+### Augmenting the catalog
+
+Once we have a clear sense of how the labels will be shared downstream, we can
+incorporate the labels themselves into the catalog database. This can be broken
+down into three steps:
+
+1. Determine which labels to use
+2. Determine an accuracy cutoff value, if any
+3. Upsert the filtered labels into the database
+
+Once step 3 is performed, the next data refresh will make the tags available in
+the API and the frontend. The specifics for each step will be determined in the
+implementation plan for this piece.
+
+## Success
+
+<!-- How do we measure the success of the project? How do we know our ideas worked? -->
+
+This project can be marked as success once the machine-generated tags from
+Rekognition are available in both the API and the frontend.
```

Comment:
I agree, although we'll have to be careful about defining it specifically as _success criteria_ -- the implication being there's some result we could observe that would make us consider reverting the project. Without having first done the project for measuring search result quality, I don't know how much confidence we can have in those measurements.

When we were discussing this as a project idea, I remember we discussed whether we should be concerned about "artificially boosting" the records that happen to be part of the Rekognition data set. Is there any way that could be harmful to search relevancy? Having more accurate tags for even a subset of data seems like it would be necessarily good, but I suppose one (maybe far-fetched) risk could be that the Rekognition-tagged records could appear with high enough frequency that a user would see the same images frequently across different searches.

This _is_ tricky to evaluate without making the search result quality measurement project a prerequisite to this one 😓 Maybe we could identify some simple worst-case scenarios that would cause us to reconsider, along the lines of @obulat's suggestion? Like if records with machine generated tags made up a certain high percentage of results across popular searches... 🤔

Reply:
I have to be honest and say that I think _any_ determination we'd like to make here regarding search result quality will be difficult to quantify before #421. Even trying to track the number of tags that show up in a high percentage of results across popular searches assumes that we have the infrastructure to be able to collate all the information necessary for that query. I'm hesitant for the reasons Staci described to add any condition onto the success criteria. While I hope that this project will improve relevancy, we don't yet have a way of assessing that. We did say something similar when [we discussed enabling iNaturalist](https://make.wordpress.org/openverse/2023/01/14/preparing-for-inaturalist/). I can try and ponder similar ways for us to mitigate it negatively impacting the search results.